HTTP_REDIRECT_ADDR=:80 breaks the AutoCert functionality (http timeout)

## What happened?

The AutoCert feature was not able to verify the http requests for issuing a valid certificate.
HTTP_REDIRECT_ADDR was set to :80
(The HTTP_REDIRECT_ADDR option is working correctly when tested on routes which already have a certificate.)

Disabling HTTP_REDIRECT_ADDR allows AutoCert to work.

## What did you expect to happen?

The Autocert feature should be able to fetch the certificates. Also when HTTP_REDIRECT_ADDR is enabled.
Otherwise the user has to manually disable redirect for every new certificate. And enable it againg afterwards.

## How'd it happen?

1. Created a new route with a new subdomain
2. Restartet pomerium
3. Waited for valid certificates

## What's your environment like?

- Pomerium version pomerium/0.10.0 (+https://wwww.pomerium.io; bfc3fb6; go1.14.7)
- Ubuntu 20.04 using latest docker

## What's your config.yaml?

```config.yaml
      - AUTHENTICATE_SERVICE_URL=https://authenticate.domain.com
      - AUTOCERT=true
      - AUTOCERT_USE_STAGING=false
      - AUTOCERT_MUST_STAPLE=true
      - AUTOCERT_DIR=/pomerium/certs/
      - IDP_PROVIDER=google
      - IDP_CLIENT_ID=someID.apps.googleusercontent.com
      - IDP_CLIENT_SECRET=SomeSECRET
      - COOKIE_SECRET=someSecret
      - SHARED_SECRET=someotherSecret
      - ADMINISTRATORS="admin@gmail.com"
      - HTTP_REDIRECT_ADDR=:80
      - POMERIUM_DEBUG=true
```

## What did you see in the logs?
Log with HTTP redirect enabled 
```logs
11:47AM INF obtaining certificate domain=some.domain.com
2020/08/20 11:47:55 [INFO][some.domain.com] Obtain certificate; acquiring lock...
2020/08/20 11:47:55 [INFO][some.domain.com] Obtain: Lock acquired; proceeding...
2020/08/20 11:47:55 [INFO][some.domain.com] Waiting on rate limiter...
2020/08/20 11:47:55 [INFO][some.domain.com] Done waiting
2020/08/20 11:47:55 [INFO] [some.domain.com] acme: Obtaining bundled SAN certificate given a CSR
2020/08/20 11:47:56 [INFO] [some.domain.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/6559538254
2020/08/20 11:47:56 [INFO] [some.domain.com] acme: Could not find solver for: tls-alpn-01
2020/08/20 11:47:56 [INFO] [some.domain.com] acme: use http-01 solver
2020/08/20 11:47:56 [INFO] [some.domain.com] acme: Trying to solve HTTP-01
2020/08/20 11:49:35 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/6559538254
2020/08/20 11:49:35 [ERROR] error: one or more domains had a problem:
[some.domain.com] acme: error: 400 :: urn:ietf:params:acme:error:connection :: Fetching http://some.domain.com/.well-known/acme-challenge/eHBjN0PY9lTku4wdeu0b9aQqeMeIdNqFKaY-oRRJum4: Timeout after connect (your server may be slow or overloaded), url: 
 (challenge=http-01 remaining=[])
2020/08/20 11:49:35 [INFO] Unable to deactivate the authorization: https://acme-v02.api.letsencrypt.org/acme/authz-v3/6559538254
2020/08/20 11:49:37 [ERROR] attempt 1: [some.domain.com] Obtain: [some.domain.com] error: one or more domains had a problem:
[some.domain.com] acme: error: 400 :: urn:ietf:params:acme:error:connection :: Fetching http://some.domain.com/.well-known/acme-challenge/eHBjN0PY9lTku4wdeu0b9aQqeMeIdNqFKaY-oRRJum4: Timeout after connect (your server may be slow or overloaded), url: 
 - retrying in 1m0s (1m42.314763847s/720h0m0s elapsed)...
      
```

Log with HTTP redirect disabled
```logs
11:52AM INF obtaining certificate domain=some.domain.com
2020/08/20 11:52:02 [INFO][some.domain.com] Obtain certificate; acquiring lock...
2020/08/20 11:52:06 [INFO][FileStorage:/pomerium/certs/] Lock for 'cert_acme_some.domain.com_acme-v02.api.letsencrypt.org-directory' is stale (created: 2020-08-20 11:47:55.412198596 +0000 UTC, last update: 2020-08-20 11:51:55.56283406 +0000 UTC); removing then retrying: /pomerium/certs/locks/cert_acme_some.domain.com_acme-v02.api.letsencrypt.org-directory.lock
2020/08/20 11:52:06 [INFO][some.domain.com] Obtain: Lock acquired; proceeding...
2020/08/20 11:52:07 [INFO] [some.domain.com] acme: Obtaining bundled SAN certificate given a CSR
2020/08/20 11:52:07 [INFO][some.domain.com] Waiting on rate limiter...
2020/08/20 11:52:07 [INFO][some.domain.com] Done waiting
2020/08/20 11:52:08 [INFO] [some.domain.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/6454245146
2020/08/20 11:52:08 [INFO] [some.domain.com] acme: Could not find solver for: tls-alpn-01
2020/08/20 11:52:08 [INFO] [some.domain.com] acme: use http-01 solver
2020/08/20 11:52:08 [INFO] [some.domain.com] acme: Trying to solve HTTP-01
2020/08/20 11:52:08 [INFO][some.domain.com] Served key authentication (HTTP challenge)
2020/08/20 11:52:08 [INFO][some.domain.com] Served key authentication (HTTP challenge)
2020/08/20 11:52:09 [INFO][some.domain.com] Served key authentication (HTTP challenge)
2020/08/20 11:52:09 [INFO][some.domain.com] Served key authentication (HTTP challenge)
2020/08/20 11:52:15 [INFO] [some.domain.com] The server validated our request
2020/08/20 11:52:15 [INFO] [some.domain.com] acme: Validations succeeded; requesting certificates
2020/08/20 11:52:17 [INFO] [some.domain.com] Server responded with a certificate.
2020/08/20 11:52:17 [INFO][some.domain.com] Certificate obtained successfully
2020/08/20 11:52:17 [INFO][some.domain.com] Obtain: Releasing lock
      
```



## Additional context

--- a/internal/autocert/manager.go
+++ b/internal/autocert/manager.go
@@ -8,6 +8,7 @@ import (
 	"net/http"
 	"sort"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/caddyserver/certmagic"
@@ -29,7 +30,7 @@ type Manager struct {
 	mu        sync.RWMutex
 	config    *config.Config
 	certmagic *certmagic.Config
-	acmeMgr   *certmagic.ACMEManager
+	acmeMgr   atomic.Value
 	srv       *http.Server
 
 	config.ChangeDispatcher
@@ -81,7 +82,7 @@ func (mgr *Manager) getCertMagicConfig(options *config.Options) (*certmagic.Conf
 	}
 	acmeMgr.DisableTLSALPNChallenge = true
 	mgr.certmagic.Issuer = acmeMgr
-	mgr.acmeMgr = acmeMgr
+	mgr.acmeMgr.Store(acmeMgr)
 
 	return mgr.certmagic, nil
 }
@@ -195,9 +196,7 @@ func (mgr *Manager) updateServer(cfg *config.Config) {
 }
 
 func (mgr *Manager) handleHTTPChallenge(w http.ResponseWriter, r *http.Request) bool {
-	mgr.mu.RLock()
-	acmeMgr := mgr.acmeMgr
-	mgr.mu.RUnlock()
+	acmeMgr := mgr.acmeMgr.Load().(*certmagic.ACMEManager)
 	if acmeMgr == nil {
 		return false
 	}

